fetch/manip.go

```diff
@@ -83,7 +84,20 @@ func Resize(src io.Reader, c *CacheContext) (io.Reader, error) {
 	}
 
 	if c.Crop {
-		options.Height = c.Width
+		data := bytes.NewReader(raw)
+
+		image, _, err := image.Decode(data)
+		if err != nil {
+			return nil, err
+		}
+
+		minDimension := int(math.Min(float64(image.Bounds().Size().X), float64(image.Bounds().Size().Y)))
+
+		if minDimension < options.Width || options.Width == 0 {
```

Comment:
This second clause works because:
- If `s=xxx` is missing from query params, then `request.FormValue("s") [returns an empty string](http://golang.org/pkg/net/http/#Request.FormValue),
- and `strconv.Atoi("")` [returns 0](http://play.golang.org/p/NI46Puj0bg).